src/backend/Device.cpp

```diff
@@ -93,25 +93,20 @@ namespace backend {
         mCaches->bindGroupLayouts.erase(obj);
     }
 
+    // Object creation API methods
+
     BindGroupBuilder* DeviceBase::CreateBindGroupBuilder() {
         return new BindGroupBuilder(this);
     }
     BindGroupLayoutBase* DeviceBase::CreateBindGroupLayout(
         const nxt::BindGroupLayoutDescriptor* descriptor) {
-        MaybeError validation = ValidateBindGroupLayoutDescriptor(this, descriptor);
-        if (validation.IsError()) {
-            // TODO(cwallez@chromium.org): Implement the WebGPU error handling mechanism.
-            delete validation.AcquireError();
-            return nullptr;
-        }
+        BindGroupLayoutBase* result = nullptr;
```

Comment:
uNit: spurious initializer? If it's never initialized by CreateBindGroupLayoutInternal(), it looks like it's never used, so should be safe to skip.

Reply:
Do you mean "if it's always initialized"?

I personally prefer to be a bit defensive about cases like this, in case we somehow end up reaching `return result` without initializing `result`. Plus I like to hope that the compiler can elide this initializer.

Reply:
I don't think the compiler can elide it if it doesn't inline the `Internal` call. It should be super super negligeable compared to the rest of object creation.